cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -215,9 +215,8 @@ public void deleteApp(HttpRequest request, HttpResponder responder,
 
       // Deletion of a particular application is not allowed if that application is used by an adapter
       if (adapterService.getApplicationTemplateInfo(appId) != null) {
-        responder.sendString(HttpResponseStatus.BAD_REQUEST,
-                             String.format("Cannot delete Application %s." +
-                                             " An AdapterType exists with a conflicting name.", appId));
+        responder.sendString(HttpResponseStatus.BAD_REQUEST, String.format(
+          "Cannot delete Application %s. An ApplicationTemplate exists with a conflicting name.", appId));
```

Comment:
Currently the name of the Application Template (which extends AbstractApplication) is same as the name of the application that is deployed. So I think the error msg is right.